core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -787,6 +787,8 @@ private[spark] class TaskSetManager(
     // SPARK-37300: when the task was already finished state, just ignore it,
     // so that there won't cause successful and tasksSuccessful wrong result.
```

Comment:
Reading this comment, the partition is already completed, probably by another `TaskSetManager`, and we just need to reset the task info here?

Reply:
I think this branch is handling a rare corner-case where the same `TaskSetManager` can mark the same task as both succeeded but failed. There is some detailed prior discussion of this in https://issues.apache.org/jira/browse/SPARK-37300